qp_shotgun/humann2/humann2.py

```diff
@@ -137,21 +159,45 @@ def humann2(qclient, job_id, parameters, out_dir):
                                                   parameters)
 
     # Step 3 execute humann2
-    commands_len = len(commands)
-    for i, cmd in enumerate(commands):
-        qclient.update_job_step(job_id, "Step 3 of 5: Executing HUMANn2"
-                                ", job %d/%d" % (i, commands_len))
-        std_out, std_err, return_value = system_call(cmd)
-        if return_value != 0:
-            error_msg = ("Error running HUMANn2:\nStd out: %s\nStd err: %s"
-                         % (std_out, std_err))
-            return False, None, error_msg
+    msg = "Step 3 of 5: Executing HUMANn2 job (%d/{0})".format(len(commands))
+    success, msg = _run_commands(qclient, job_id, commands, msg)
+    if not success:
+        return False, None, msg
 
     # Step 4 merge tables
-    qclient.update_job_step(job_id, "Step 4 of 5: Merging resulting tables")
-
-    # Step 5 generating re-normalized tables: TODO
-    qclient.update_job_step(job_id, "Step 5 of 5: Re-normalizing tables")
-    artifacts_info = []
-
-    return True, artifacts_info, ""
+    commands = []
+    commands.append(('humann2_join_tables -i {0} -o {0}/genefamilies.biom '
```

Comment:
Note that these are going to be by default the funny stratified tables, so that there will be a line for the gene family and then a bunch of lines for that same gene family split across each of the different genomes that had hits contributing to it. (in other words, each annotated read gets represented twice per table)

We don't have a lot of tools to deal with this data structure right now -- maybe want to include the --remove-stratified-output flag to have simple count tables for now.